src/gt4py/next/program_processors/runners/dace_iterator/itir_to_sdfg.py

```diff
@@ -432,24 +431,40 @@ def _visit_scan_stencil_closure(
                     name, dtype=as_dace_type(cast(ts.ScalarType, self.storage_types[name]))
                 )
 
+        connectivity_arrays = [(scan_sdfg.arrays[name], name) for name in connectivity_names]
+
         # implement the lambda closure as a nested SDFG that computes a single item of the map domain
         lambda_context, lambda_inputs, lambda_outputs = closure_to_tasklet_sdfg(
             node,
             self.offset_provider,
             {},
             [],
-            [],
+            connectivity_arrays,
             self.node_types,
         )
 
-        input_connectors = {
-            inner_name for inner_name, expr in lambda_inputs if inner_name not in scan_sdfg.symbols
-        }
+        connectivity_access_nodes = [lambda_state.add_access(name) for name in connectivity_names]
+
+        # Map symbols by matching outer and inner strides, shapes, while defaulting to same symbol
+        symbol_mapping = {sym: sym for sym in lambda_context.body.free_symbols}
+        for inner_name, access_node in zip(connectivity_names, connectivity_access_nodes):
+            outer_data = scan_sdfg.arrays[access_node.data]
+            inner_data = lambda_context.body.arrays[inner_name]
+            for o_sh, i_sh in zip(outer_data.shape, inner_data.shape):
+                if str(i_sh) in lambda_context.body.free_symbols:
+                    symbol_mapping[str(i_sh)] = o_sh
+            for o_sh, i_sh in zip(outer_data.strides, inner_data.strides):
+                if str(i_sh) in lambda_context.body.free_symbols:
+                    symbol_mapping[str(i_sh)] = o_sh
```

Comment:
Why do we only have to do this for connectivities, but not all other arrays?

Reply:
The output of a lambda seems to be a scalar, so not needed. The inputs of a lambda seem to be scalars and symbols. From the examples I have seen, the scan lambda behaves as a tasklet without side effects. Input data is passed as scalars (by value), while the memlet takes care of addressing the right element in the input array.
If this is the case, I guess the next question is why do we need to add connectivities to the scan lambda sdfg at all. The answer is that I was trying to address a runtime error in `visit_Lambda`, which is a generic function and should work for any kind of lambda function.
```
    def visit_Lambda(
        self, node: itir.Lambda, args: Sequence[ValueExpr | SymbolExpr]
    ):
        neighbor_tables = filter_neighbor_tables(self.offset_provider)
        param_names = [str(p.id) for p in node.params]
        conn_names = [connectivity_identifier(offset) for offset, _ in neighbor_tables]
```
Since we are passing the connectivity arrays, I can deduce that a lambda can access array elements directly (by reference) in the generic case.